libqtile/widget/base.py

```diff
@@ -891,6 +890,9 @@ def _configure(self, qtile, bar):
         # there's no reflection, the mirror matches its parent bar.
         self.drawer.clear(self.background or self.bar.background)
 
+    def calculate_length(self):
+        return self.reflects.calculate_length()
+
```

Comment:
If here it's always using the length of the widget it reflects, how does it stretch according to its own bar?

Reply:
Ah of course! :smile: 

Are there any weird or unexpected effects that could arise by having mirrored widgets be different sizes? I suppose the only risk is that the content of a larger original widget can be cut off when drawing to a smaller mirror? Is drawing a larger buffer to a smaller buffer allowed?

Reply:
The only time the widget would be a different size is when it's a stretch widget. In that scenario, contents could be cut off if the widget has less space. This is fine and is no different to the default behaviour of a stretch widget.